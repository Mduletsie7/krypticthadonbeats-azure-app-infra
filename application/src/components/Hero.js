import React from "react";
import "../styles/Hero.css";

const Hero = () => {
  return (
    <section className="hero" role="banner" aria-label="Hero section featuring Kryptic Tha Don">
      <div className="hero-container">
        <div className="hero-content">
          <header className="hero-text">
            <h1>
              <span className="hero-brand">Kryptic</span> Tha Don
            </h1>
            <p className="hero-description">
              Music Producer from Cape Town, South Africa with a passion for Hip-Hop.
            </p>
          </header>
        </div>
      </div>
    </section>
  );
};

export default Hero;