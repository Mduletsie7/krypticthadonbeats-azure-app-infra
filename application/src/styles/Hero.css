/* HeroSection.css */
.hero {
  /* background-image: url('https://krypticmusicportfolio.s3.amazonaws.com/images/websitehero.jpg'); */
  background-size: cover; /* Cover the entire container */
  background-repeat: no-repeat;
  background-position: center; /* Center the background image */
  background-color: burlywood; /* Adjust the last value (0.5) for opacity */
    color: black;
    height: 100vh;
    display: flex;
    flex-direction: column; /* Updated to column layout */
    justify-content: center;
    align-items: center;
    text-align: center;
  }
  
  .hero-content {
    display: flex;
    flex-direction: column; /* Updated to column layout */
    align-items: center;
  }
  
  .hero-image {
    max-width: 100%;
    max-height: 300px; /* Set the maximum height as per your design */
    margin-bottom: 20px; /* Added margin to create space between image and text */
    border-radius: 8px; /* Optional: Add border-radius for a rounded look */
  }
  
  .hero-text {
    text-align: center;
  }
  .herotxt-kryptic {
    color: #000;
  
  }
  
  .hero-text h1 {
    color: #000;
    font-size: 5rem;
    margin-bottom: 20px;
  }
  
  .hero-text p {
    font-size: 1.5rem;
    margin-bottom: 30px;
  }
  
  .cta-button {
    background-color: whitesmoke;
    color: black;
    padding: 15px 30px;
    font-size: 1.2rem;
    border: none;
    cursor: pointer;
    border-radius: 5px;
  }
  
  .cta-button:hover {
    background-color: #36454f;
  }
  
  /* Add any additional styles to customize your hero section */
  