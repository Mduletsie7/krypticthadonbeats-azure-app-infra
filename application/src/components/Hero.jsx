// HeroSection.js
import React from "react";
import "../styles/Hero.css";
// import image from "../assets/kryptic.png";

const Hero = () => {
  return (
    <section className="hero">
      <div className="hero-content">
        <div className="hero-text">
          <h1><span className="herotxt-kryptic">Kryptic</span> Tha Don</h1>
          <p>
            Music Producer from Cape Town, South africa with a passion for Hip-Hop.
          </p>
        </div>
      </div>
    </section>
  );
};

export default Hero;
