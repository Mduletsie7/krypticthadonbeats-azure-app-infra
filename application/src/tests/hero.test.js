// HeroSection.test.js
import React from "react";
import { render, screen } from "@testing-library/react";
import Hero from "../components/Hero"; // Adjust the path if needed
import '@testing-library/jest-dom';

describe("Hero Section", () => {
  beforeEach(() => {
    render(<Hero />);
  });

  it("renders the Hero section", () => {
    const section = screen.getByRole("region", { name: /hero section/i });
    expect(section).toBeInTheDocument();
  });

  it("displays the correct heading", () => {
    const heading = screen.getByRole("heading", { name: /kryptic tha don/i });
    expect(heading).toBeInTheDocument();
  });

  it("displays the correct description", () => {
    const paragraph = screen.getByText(/music producer from Cape Town/i);
    expect(paragraph).toBeInTheDocument();
  });

  it("has a styled span with class 'herotxt-kryptic'", () => {
    const span = screen.getByText("Kryptic");
    expect(span).toHaveClass("herotxt-kryptic");
  });
});
