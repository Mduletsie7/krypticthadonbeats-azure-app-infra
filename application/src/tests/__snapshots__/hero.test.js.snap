// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component Snapshots matches snapshot 1`] = `
<section
  aria-label="Hero Section"
  class="hero"
  role="region"
>
  <div
    class="hero-content"
  >
    <div
      class="hero-text"
    >
      <h1>
        <span
          class="herotxt-kryptic"
        >
          Kryptic
        </span>
         Tha Don
      </h1>
      <p>
        Music Producer from Cape Town, South africa with a passion for Hip-Hop.
      </p>
    </div>
  </div>
</section>
`;
